File: apps/ios/Packages/Model/Sources/Model/status/MediaRequestStatus.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
// Seerr server/constants/media.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
public enum MediaRequestStatus: Sendable, Equatable, Codable {
    case pending
    case approved
    case declined
    case failed
    case completed
    case unrecognized(Int?)

    public init(rawValue: Int?) {
        switch rawValue {
        case 1: self = .pending
        case 2: self = .approved
        case 3: self = .declined
        case 4: self = .failed
        case 5: self = .completed
        default: self = .unrecognized(rawValue)
        }
    }

    public var rawValue: Int? {
        switch self {
        case .pending: 1
        case .approved: 2
        case .declined: 3
        case .failed: 4
        case .completed: 5
        case .unrecognized(let raw): raw
        }
    }
}

File: apps/android/core/model/src/main/kotlin/app/gauja/core/model/status/MediaRequestStatus.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.model.status

// Seerr server/constants/media.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
sealed class MediaRequestStatus(val rawValue: Int?) {
    data object PENDING : MediaRequestStatus(1)

    data object APPROVED : MediaRequestStatus(2)

    data object DECLINED : MediaRequestStatus(3)

    data object FAILED : MediaRequestStatus(4)

    data object COMPLETED : MediaRequestStatus(5)

    data class Unrecognized(val value: Int?) : MediaRequestStatus(value)

    companion object {
        fun fromWire(raw: Int?): MediaRequestStatus =
            when (raw) {
                1 -> PENDING
                2 -> APPROVED
                3 -> DECLINED
                4 -> FAILED
                5 -> COMPLETED
                else -> Unrecognized(raw)
            }
    }
}
